pkg/resource/deploy/step_generator.go

```diff
@@ -38,6 +39,78 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/result"
 )
 
+// ContinueResourceDiffEvent is a step that asks the engine to continue provisioning a resource after completing its
+// diff, it is always created from a base RegisterResourceEvent.
+type ContinueResourceDiffEvent interface {
+	Event() RegisterResourceEvent
+	Error() error
+	Diff() plugin.DiffResult
+	URN() resource.URN
+	Old() *resource.State
+	New() *resource.State
+	Goal() *resource.Goal
+	Provider() plugin.Provider
+	Autonaming() *plugin.AutonamingOptions
+	RandomSeed() []byte
```

Comment:
We should doc comment these. Also, doesn't the base RRE have some of these? Not sure if you'd want to embed the RRE interface for those as a result and add the new bits?